rest/config.go

```diff
@@ -676,6 +677,38 @@ func (config *ServerConfig) setupAndValidateDatabases() (errs error) {
 	return nil
 }
 
+// expandEnv replaces $var or ${var} in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by an empty string. A default value can
+// be given by using the form ${var:-default value}.
+func expandEnv(config []byte) []byte {
+	return []byte(os.Expand(string(config), func(key string) string {
+		if key == "$" {
+			base.Debugf(base.KeyAll, "Skipping environment variable expansion: %s", key)
+			return key
+		}
+		return envDefaultExpansion(key, os.Getenv)
+	}))
+}
+
+// envDefaultExpansion implements the ${foo:-bar} parameter expansion from
+// https://pubs.opengroup.org/onlinepubs/009695399/utilities/xcu_chap02.html#tag_02_06_02
+func envDefaultExpansion(key string, getEnvFn func(string) string) (value string) {
+	kvPair := strings.SplitN(key, ":-", 2)
+	key = kvPair[0]
+	value = getEnvFn(key)
+	if value == "" && len(kvPair) == 2 {
+		// Set value to the default.
+		value = kvPair[1]
+		base.Debugf(base.KeyAll, "Replacing config environment variable '${%s}' with "+
```

Comment:
As discussed, shouldn't log values as they might be credentials.  However, would be useful to log at info level if the value is empty string and there's no default set, as a diagnostic for unset environment variables.

Reply:
Sure, that makes sense. Removed the changes to log redacted environment variable values and added changes to log debug level warning for each variable replacement and info level warning when the value is empty string and there's no default set.